salt/cloud/clouds/vmware.py

```diff
@@ -656,10 +656,9 @@ def _manage_devices(devices, vm=None, container_ref=None, new_vm_name=None):
                     if device.deviceInfo.label in list(devices['disk'].keys()):
                         disk_spec = None
                         if 'size' in devices['disk'][device.deviceInfo.label]:
-                            disk_spec = _get_size_spec(device, devices)
-                            size_kb = float(
-                                devices['disk'][device.deviceInfo.label]['size']
-                            ) * 1024 * 1024
+                            size_gb = float(devices['disk'][disk_label]['size'])
```

Comment:
I must have been in another dimension yesterday, because that will clearly raise a NameError and should instead be `device.deviceInfo.label`